src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -463,15 +479,25 @@ async function _installDebugger(
     try {
         const vsdbgPath = getDebuggerPath(targetFolder)
 
+        const samCliContext = getSamCliContext()
+        const samCliVersionValidatorResult = await samCliContext.validator.getVersionValidatorResult()
+        const samCliVersion = samCliVersionValidatorResult.version
+
+        const imageStr =
+            !samCliVersion || semver.gte(samCliVersion, '1.0.0')
+                ? `amazon/aws-sam-cli-emulation-image-${lambdaRuntime}`
+                : `lambci/lambda:${lambdaRuntime}`
```

Comment:
this logic should probably live in one of the SamCli things (e.g. `SamCliContext.getDockerImageName()`). That also will reduce friction when merging to `feature/debugconfig`